### PR TITLE
tweets_hate_speech_detection

### PR DESCRIPTION
**Pull request by darshan-gandhi**

### CODE REVIEW

**Comment by darshan-gandhi** [comment]
Hi @lhoestq  I have added this new dataset for tweet's hate speech detection. 

Please if u could review it. 

Thank you

File: datasets/hatespeech/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/hatespeech/README.md:2]:
```suggestion
```

File: datasets/hatespeech/hatespeech.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+class HateSpeech(datasets.GeneratorBasedBuilder):
+    """Detecing which tweets showcase hate or racist remarks."""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "label": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/hatespeech/hatespeech.py:54]: Can you use the ClassLabel feature type here ? 
For example
```
"label": datasets.ClassLabel(names=["no-hate-speech", "hate-speech"])
```

File: datasets/hatespeech/hatespeech.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            csv_reader = csv.reader(
+                csv_file, quotechar='"', delimiter=",", quoting=csv.QUOTE_ALL, skipinitialspace=True
+            )
+            next(csv_reader, None)
+            for id_, row in enumerate(csv_reader):
+                row = row[1:]
+                (label, tweet) = row
+
+                yield id_, {
+                    "label": (label),
```
Comment (lhoestq) [inline datasets/hatespeech/hatespeech.py:81]:
```suggestion
                    "label": int(label),
```
can you convert the label to integer to make it work with ClassLabel ?

File: datasets/hatespeech/README.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+## Considerations for Using the Data
+
+### Social Impact of Dataset
+
+With the help of this dataset, one can understand more about the human sentiments and also analye the situations when a particular person intends to make use of   hatred/racist comments 
+
+### Discussion of Biases
+
+The data could be cleaned up further for additional purposes such as applying a better feature extraction techniques
```
Comment (lhoestq) [inline datasets/hatespeech/README.md:125]:
```suggestion
The data could be cleaned up further for additional purposes such as applying a better feature extraction techniques

`[More Information Needed]`
```

**Review by lhoestq — COMMENTED** [review]
Awesome thanks :) 

I left a few comments

Can you also rename the dataset `twitter_hatespeech` instead of `hatespeech` which is too generic ? To do so please rename both the directory name and the dataset script name. Then you'll also have to rename the class name `TwitterHatespeech`.

After all the changes you'll have to regenerate the dataset_infos.json file

**Review by B-starboy — APPROVED** [review]

**Comment by lhoestq** [comment]
Hi @darshan-gandhi have you add a chance to take a look at my suggestions ?

Feel free to ping me when you're ready for the final review

**Comment by lhoestq** [comment]
Closing in favor of #1607

**Pull request closed without merging**